File: cookbook/php/createField.php
<?php

//Ici, renseignez la xKey et les parametres personnalises du message
$xKey = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
$fieldId = 0123;	//Id du critere a modifier ('null' si le critere est a creer)

$type = 'numeric';	//Code pour creer un critere :
	//Types : email = 'email' / telephone = 'phone' / zone de texte = 'textArena' / chaine de caractere = 'textField' / valeur numerique = 'numeric' / date = 'date'
	//Listes : liste de valeurs = 'singleSelectList' / Liste de valeurs multiples = 'multipleSelectList'
$name = 'createField (php)';	//Nom du critere
$isUnicity = false;	//Unicite : oui = 'true' / non = 'false'
$isMandatory = true;	//Obligatoire : oui = 'true' / non = 'false'

$constraintOperator = 1;	//Pour les valeurs numeriques et les dates : 1 = '>' / 2 = '<' / 3 = '>=' / 4 = '<=' / 5 = '=' ('null' pour rien)
$constraintValue = "42";	//Valeur de la contrainte ex : ["42"] ('null' pour rien)

$valueListId = null;	//Id de la liste dans le cas d'un 'singelSelectList' ou un 'multipleSelectList' ('null' pour rien)

//On trouve l'addresse pour la requete
$url = 'http://v8.mailperformance.com/fields/' . $fieldId;

//Creation du Json du message
if ($constraintOperator != null && $constraintValue != null)
{
	$arr = array(
			'type' => $type,
			'name' => $name,
			'isUnicity' => $isUnicity,
			'isMandatory' => $isMandatory,
			'constraint' => array(
					'operator' => $constraintOperator,
					'value' => $constraintValue));
	
}
else if ($valueListId != null)
{
	$arr = array(
			'type' => $type,
			'name' => $name,
			'isUnicity' => $isUnicity,
			'isMandatory' => $isMandatory,
			'valueListId' => $valueListId);
}
else
{
	$arr = array(
			'type' => $type,
			'name' => $name,
			'isUnicity' => $isUnicity,
			'isMandatory' => $isMandatory);
}

//On affiche le message
$message = json_encode($arr);
echo $message . "\n";

//Connexion
if ($fieldId != null)
	$con = connect($url, $xKey, $message, 'PUT');
else
	$con = connect($url, $xKey, $message, 'POST');

$result = $con['result'];
$info = $con['info'];
$req = $con['req'];

if ($info['http_code'] != 200)
{
	echo 'Error : ' . $info['http_code'];
}
else
{
	//Le critere a bien ete cree
	echo 'The field  : ' . $name . " is created / update.\n\n";
}
curl_close($req);



//Fonctions -----



//Utilisation de cURL pour remplir les requetes
function startCurlInit($url)
{
	$init = curl_init();
	curl_setopt($init, CURLOPT_URL, $url);
	curl_setopt($init, CURLOPT_RETURNTRANSFER, true);
	return ($init);
}

//Fonction de connexion
function connect($url, $xKey, $message, $method)
{
	//On remplit la requete
	$req = startCurlInit($url);
	curl_setopt($req, CURLOPT_CUSTOMREQUEST, $method);
		
	//Mise en place du xKey et des options
	curl_setopt($req, CURLOPT_HTTPHEADER, array(
	'X-Key: ' . $xKey,
	'Content-Type: application/json',
	'Content-Length: ' . strlen($message)));
	curl_setopt($req, CURLOPT_POSTFIELDS, $message);

	//Execution de la requete
	$result = curl_exec($req);

	//Verification des reponses
	$info = curl_getinfo($req);
	
	return (array('result' => $result, 'info' => $info, 'req' => $req));
}

?>

File: Csharp/createField.cs
using Newtonsoft.Json.Linq;
using System;
using System.IO;
using System.Net;
using System.Xml;

namespace createField
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            // Changez le Path pour correspondre à la destination de votre fichier de configuration
            var doc = new XmlDocument();
            doc.Load("config.xml");

            var xKey = doc.DocumentElement.SelectSingleNode("/config/xKey").InnerText;
            var baseUrl = doc.DocumentElement.SelectSingleNode("/config/url").InnerText;
            
            string fieldId = null;	//Id du critere a modifier ('null' si le critere doit etre cree)

            var type = "numeric";	//Code pour creer un critere :
				//Normales : email = 'email' / telephone = 'phone' / zone de texte = 'textArena' / chaine de caractere = 'textField' / valeur numerique = 'numeric' / date = 'date'
				//Listes : liste de valeurs = 'singleSelectList' / Liste de valeurs multiples = 'multipleSelectList'
            var name = "createField (Csharp)";	//Nom du critere
            var isUnicity = false;	//Unicite : oui = 'true' / non = 'false'
            var isMandatory = false;	//Obligatoire : oui = 'true' / non = 'false'

            var constraintOperator = 0;	//Pour les valeurs numeriques et les dates : 1 = '>' / 2 = '<' / 3 = '>=' / 4 = '<=' / 5 = '==' ('0' pour rien)
            var constraintValue = "42";	//Valeur de la contrainte ('null' pour rien)

            var valueListId = 0;	//Id de la liste dans le cas d'un 'singleSelectList' ou un 'multipleSelectList' ('0' pour rien)


            //On trouve l'adresse pour la requete
            var url = baseUrl + "fields/" + fieldId;
            //Creation du Json du message
            var jsonData = new JObject();
            jsonData.Add("type", type);
            jsonData.Add("name", name);
            jsonData.Add("isUnicity", isUnicity);
            jsonData.Add("isMandatory", isMandatory);

            if (constraintOperator != 0 && constraintValue != null)
            {
                var constraint = new JObject();
                constraint.Add("operator", constraintOperator);
                constraint.Add("value", constraintValue);
                jsonData.Add("constraint", constraint);
            }
            else if (valueListId != 0)
            {
                jsonData.Add("valueListId", valueListId);
            }

            //On affiche le message
            Console.Write(jsonData + "\n");


            //Lancement de la connexion pour remplir la requete
            var connection = Utils.allConnection(url, xKey, jsonData, (segmentId == null ? "POST" : "PUT"));

            var response = (int)connection[0];
            var con = (HttpWebRequest)connection[1];
            var httpResponse = (HttpWebResponse)connection[2];
            var responseString = (string)connection[3];

            //Verification des reponses
            if (response != 200)
            {
                //Affichage de l'erreur
                Console.Write("\nError : " + response + "\n\n");
            }
            else
            {
                //Le critere a bien ete cree
                Console.Write("The field  : " + name + " is created / update.");
            }
            //Attente de lecture (optionnel)
            Console.ReadLine();

            httpResponse.Close();
        }
    }
}
